### ISSUE TITLE
[Linux/Wine] Regression on recent build 5e42d18, some games are crashing at startup

### ISSUE THREAD

**Issue by lorn10** [Cxbx-Reloaded/Cxbx-Reloaded#2352]
## Quick summary

It looks that build 5e42d18 has introduced a new problem for Linux/Wine users. Some games are no longer working, they crash at boot. 
The former build https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/commit/374ba5ec703e22d012c6c7f0f477914316426c23 and https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/commit/696d49820c5293beff093d73e4fae17ff56f3859 are working fine.

## Details

When I start up the game **Crash Bandicoot: The Wrath of Cortex** or **Baldur's Gate: Dark Alliance** I get a crash. See the backtrace files for more information.

[backtrace_cxbx_5e42d18_BaldursGate.txt](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/files/8598747/backtrace_cxbx_5e42d18_BaldursGate.txt)
[backtrace_cxbx_5e42d18_CrashBandicoot.txt](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/files/8598748/backtrace_cxbx_5e42d18_CrashBandicoot.txt)

Note, **Crash Tag Team Racing** is working fine. But when I stop the game via the **stop** button then the CXBX-R hangs. I have to kill it and restart the whole application.

### System Configuration:

```
test@test-iMac:~$ inxi -b
System:    Host: test-iMac Kernel: 5.13.0-40-generic x86_64 bits: 64 Desktop: KDE Plasma 5.18.8 
           Distro: Ubuntu 20.04.4 LTS (Focal Fossa) 
Machine:   Type: Desktop System: Apple product: iMac12,2 v: 1.0 serial: <superuser/root required> 
           Mobo: Apple model: Mac-942B59F58194171B v: iMac12,2 serial: <superuser/root required> UEFI: Apple 
           v: IM121.88Z.0047.B1F.1201241648 date: 01/24/12 
CPU:       Quad Core: Intel Core i5-2500S type: MCP speed: 1597 MHz min/max: 1600/3700 MHz 
Graphics:  Device-1: Intel 2nd Generation Core Processor Family Integrated Graphics driver: i915 v: kernel 
           Device-2: Advanced Micro Devices [AMD/ATI] Whistler [Radeon HD 6730M/6770M/7690M XT] driver: radeon v: kernel 
           Display: x11 server: X.Org 1.20.13 driver: modesetting,radeon FAILED: ati unloaded: fbdev,vesa 
           resolution: 2560x1440~60Hz 
           OpenGL: renderer: AMD TURKS (DRM 2.50.0 / 5.13.0-40-generic LLVM 14.0.0) 
           v: 3.3 Mesa 22.2.0-devel (git-14cad38 2022-04-30 focal-oibaf-ppa) 
Network:   Device-1: Broadcom and subsidiaries NetXtreme BCM57765 Gigabit Ethernet PCIe driver: tg3 
           Device-2: Qualcomm Atheros AR93xx Wireless Network Adapter driver: ath9k 
Drives:    Local Storage: total: 961.01 GiB used: 161.13 GiB (16.8%) 
Info:      Processes: 214 Uptime: 2d 4h 04m Memory: 7.74 GiB used: 2.30 GiB (29.7%) Shell: bash inxi: 3.0.38 
```

### Additional Information (if any):

The corresponding wine prefix has installed the required `msvcp140_atomic_wait.dll`. I have tested this also with a prefix which contains the whole `vcrun2019`. With the same results, - a crash at startup. However, because build https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/commit/374ba5ec703e22d012c6c7f0f477914316426c23 is working fine, this has most likely no relation with the `msvcp140_atomic_wait.dll` matter.

**Comment by RadWolfie** [comment]
You are using Wine 7.7 staging build which is not acceptable in our ticket system. Please use stable Wine 7.0 release before filing a report.

> Note, Crash Tag Team Racing is working fine. But when I stop the game via the stop button then CXBX-R hangs. I have to kill it and restart the whole application.

I can confirm the problem occur in my vm and only once on Windows with JSRF title. Reproduce this problem isn't easy to catch.

**Comment by lorn10** [comment]
Okay, I have one computer which uses Wine 7.0 stable, will check if it happens also there. 

If not, I will close the issue here again. And by the way, Wine 7.7 is described as "devel" not "staging". The later also exists as a separate third branch which has effectively then a "nightly" or "experimental" status while the "devel" branch can be regarded as "semi-stable". :wink:

**Comment by lorn10** [comment]
Well, instead of crashing the mentioned two games are just **hanging** / **not loading** on startup under Wine 7.0 and build https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/commit/5e42d181f2daf4a244450b1535223d4c71ea5e54. 

With build https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/commit/696d49820c5293beff093d73e4fae17ff56f3859 both are running fine and that on a **Radeon HD 2400XT** super low-end GPU with just **128MB** of VRAM. :smile:  
The mentioned iMac8,1 has the following specs:

```
test@iMac-test:~$ inxi -b
System:    Host: iMac-test Kernel: 5.13.0-40-generic x86_64 bits: 64 Desktop: KDE Plasma 5.18.8 
           Distro: Ubuntu 20.04.4 LTS (Focal Fossa) 
Machine:   Type: Desktop System: Apple product: iMac8,1 v: 1.0 serial: <superuser/root required> 
           Mobo: Apple model: Mac-F226BEC8 v: PVT serial: <superuser/root required> UEFI: Apple 
           v: IM81.88Z.00C1.B00.0802091538 date: 02/09/08 
CPU:       Dual Core: Intel Core2 Duo E8135 type: MCP speed: 798 MHz min/max: 800/2400 MHz 
Graphics:  Device-1: Advanced Micro Devices [AMD/ATI] RV610/M74 [Mobility Radeon HD 2400 XT] driver: radeon v: kernel 
           Display: x11 server: X.Org 1.20.13 driver: radeon FAILED: ati unloaded: fbdev,modesetting,vesa 
           resolution: 1680x1050~60Hz 
           OpenGL: renderer: AMD RV610 (DRM 2.50.0 / 5.13.0-40-generic LLVM 12.0.0) v: 3.3 Mesa 21.2.6 
Network:   Device-1: Broadcom and subsidiaries BCM4360 802.11ac Wireless Network Adapter driver: wl 
           Device-2: Marvell 88E8058 PCI-E Gigabit Ethernet driver: sky2 
Drives:    Local Storage: total: 262.39 GiB used: 86.83 GiB (33.1%) 
Info:      Processes: 185 Uptime: 55m Memory: 3.83 GiB used: 966.8 MiB (24.6%) Shell: bash inxi: 3.0.38 
```

**Comment by lorn10** [comment]
A new status update, the topic changed somewhat with build 1077115, so I can load now most games normally but the hanger when **stop emulation** is selected is still there. Note, this is now reproducible under Wine on **every game**.

So unfortunately it looks that #2355 didn't help much in that regard.

**Comment by lorn10** [comment]
Okay, it looks that 95b789e runs again almost fully error free under Wine. :+1:

Will observe the situation on the next builds and then close this issue if the "stop emulation bug" really no longer occurs.

**Comment by RadWolfie** [comment]
"stop emulation" deadlock will continue to occur as it is a separate issue.

**Comment by lorn10** [comment]
@RadWolfie  
A final addition to this topic. Regarding build 95b789e and **Wine 7.9** the "stop emulation" issue does no longer occurs always on every game. The issue may be still present but it occurs much less frequently.
